src/compiler/checker.ts

```diff
@@ -20217,19 +20217,25 @@ namespace ts {
             return checkPropertyAccessExpressionOrQualifiedName(node, node.left, node.right);
         }
 
+        function isMethodAccessForCall(node: Node) {
+            while (node.parent.kind === SyntaxKind.ParenthesizedExpression) {

```

Comment:
This should probably? also skip both kinds of casts.

Reply:
No, if there is an intervening type assertion then that specifies the type and widening isn't relevant.